data-otservbr-global/scripts/actions/other/fishing.lua

```diff
@@ -81,7 +81,7 @@ function fishing.onUse(player, item, fromPosition, target, toPosition, isHotkey)
 		return true
 	end
 
-	if useWorms and player:removeItem("worm", 1) and targetId == 21414 then
+	if useWorms and targetId == 21414 and player:removeItem("worm", 1) then
```

Comment:
Just trying to understand, `player:removeItem("worm", 1)` executes the functions remove item, and because of the order it was running even if the targetId was not water?

Reply:
exactly!